workflow-engine/src/main/java/com/redhat/parodos/workflows/workflow/ParallelFlowReport.java

```diff
@@ -86,12 +86,16 @@ void addAll(List<WorkReport> workReports) {
 	 */
 	@Override
 	public WorkStatus getStatus() {
+		WorkStatus workStatus = WorkStatus.COMPLETED;
 		for (WorkReport report : reports) {
 			if (report.getStatus().equals(WorkStatus.FAILED)) {
 				return WorkStatus.FAILED;
 			}
+			if (!report.getStatus().equals(WorkStatus.COMPLETED)) {
+				workStatus = report.getStatus();
```

Comment:
I think the parallel workflow should be in_progress if any of its works is in_progress ? the logic here seems to set workStatus as of the last item in reports?